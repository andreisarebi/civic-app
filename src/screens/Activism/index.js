import React from 'react';
import { Share, ScrollView, View, Text, StyleSheet, Button } from 'react-native';
import PropTypes from 'prop-types';
import SquareButton from './SquareButton';
import { Icon } from 'react-native-elements';
import ShareButton from '../components/ShareButton';
import Colors from '../../styles/colors';

class ActivismScreen extends React.Component {
  static navigationOptions = {
    title: 'Activism',
  };

  handlePress = () => {
    Share.share({
      message: "Check out the Civic App!",
      url: "https://www.getcivicapp.com/",
      title: "Civic App"
    },
    {
      dialogTitle: "Civic App"
    })
  }

  render() {
    const { handlePress } = this;
    return (
      <View style={styles.container}>
        <View style={styles.card}>
          <Text style={styles.heading}>Help friends vote and boost your impact</Text>
          <Text style={styles.subHeading}>Invite friends in California to vote with a personal note!</Text>
          <Text style={styles.subHeading}>Research shows that personal messages increase voter turnout</Text>
          <Icon
            name="megaphone"
            size={90}
            type="entypo"
            color={Colors.darkBlue}
          />
          <View style={styles.containerViewStyle}>
            <ShareButton
              title="Invite Friends with Civic"
              buttonStyle={styles.buttonStyle}
              onPress={handlePress}
            />
          </View>
        </View>
      </View>
    );
  }
}

ActivismScreen.propTypes = {
  user: PropTypes.string
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 20,
    alignItems: 'center'
  },
  card: {
    flex:1,
    backgroundColor: Colors.white,
    borderRadius: 20,
    flexDirection: 'column',
    justifyContent: 'space-between',
    marginLeft: 10,
    marginRight: 10,
    maxHeight: 480,
    maxWidth: 330,
    paddingTop: 20,
    paddingBottom: 20,
    paddingLeft: 20,
    paddingRight: 20,
    width: '100%',
    elevation:2,
    shadowColor:'black',
    shadowOffset: {
      width:2, height:2
    },
    shadowRadius: 3,
    shadowOpacity: 0.2
  },
  heading: {
    fontSize: 28,
    textAlign: 'center',
    margin: 10,
    marginBottom: 0
  },
  subHeading: {
    fontSize: 18,
    margin: 10,
    marginTop: 0,
    marginBottom: 0
  },
  containerViewStyle: {
    margin: 10,
  },
  buttonStyle: {
    justifyContent: 'space-around',
    backgroundColor: Colors.red,
  }
})

export default ActivismScreen;
