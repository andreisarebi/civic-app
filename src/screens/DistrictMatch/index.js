import React, {Component} from 'react';
import PropTypes from 'prop-types';
import { View, Button } from 'react-native';
import DistrictMatch from './DistrictMatch';

import styles from '../styles';

class DistrictMatchScreen extends Component {
  render() {
    return (
      <View style={styles.container}>
        <DistrictMatch navigate={this.props.navigation.navigate}/>
      </View>
    );
  }
}

DistrictMatchScreen.propTypes = {
  navigation: PropTypes.objectOf({
    navigate: PropTypes.func,
    push: PropTypes.func,
  }),
};

export default DistrictMatchScreen;
