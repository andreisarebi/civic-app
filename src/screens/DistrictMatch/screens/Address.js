import React, { Fragment } from 'react';
import { View, Text, TextInput, Image } from 'react-native';
import ButtonSubmit from '../../components/ButtonSubmit';
import PropTypes from 'prop-types';

const addressImage = require('../../../assets/images/district-match-address.png');

const Address = ({styles, address, city, error, handleChange, submit}) => {
  return(
    <Fragment>
      <Text style={styles.bodyText}>Oops! Your zip code spans several districts. To help us identify your district, please enter your home address.</Text>
      <Image style={styles.image} source={addressImage}/>
      <View style={styles.inputContainer}>
        <TextInput
          autoCapitalize="none"
          autoCorrect={false}
          autoFocus={false}
          clearButtonMode="while-editing"
          keyboardType="default"
          style={styles.textInput}
          placeholder="Street Address"
          onChangeText={(value)=>handleChange({address:value})}
          value={address}
        />
        <TextInput
          autoCapitalize="none"
          autoCorrect={false}
          autoFocus={false}
          clearButtonMode="while-editing"
          keyboardType="default"
          style={styles.textInput}
          placeholder="City"
          onChangeText={(value)=>handleChange({city:value})}
          value={city}
        />
        {
          <View>
            <Text style={styles.errorText}>{error && error}</Text>
          </View>
        }
        <ButtonSubmit onPress={submit}/>
      </View>
    </Fragment>
  )
};

Address.propTypes = {
  styles: PropTypes.object,
  address: PropTypes.string,
  city: PropTypes.string,
  error: PropTypes.string,
  handleChange: PropTypes.func,
  submit: PropTypes.func
}

export default Address;
