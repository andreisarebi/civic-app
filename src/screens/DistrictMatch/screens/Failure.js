import React, { Fragment } from 'react';
import { Text, Button } from 'react-native';
import PropTypes from 'prop-types';

const Failure = ({navigate}) => (
  <Fragment>
    <Text>We Couldn't Find Your District :( </Text>
    <Button title="Go to Elections" onPress={()=>navigate('Elections')} />
  </Fragment>
);

Failure.propTypes = {
  navigate: PropTypes.func
};

export default Failure;
