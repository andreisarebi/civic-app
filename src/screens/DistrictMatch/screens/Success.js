import React, { Fragment } from 'react';
import { Text, Button } from 'react-native';
import PropTypes from 'prop-types';

const Success = ({district, navigate}) => (
  <Fragment>
    <Text>Your District Is: {district}</Text>
    <Button title="Go to Elections" onPress={()=>navigate('Elections')} />
  </Fragment>
);

Success.propTypes = {
  district: PropTypes.string,
  navigate: PropTypes.func
};

export default Success;
