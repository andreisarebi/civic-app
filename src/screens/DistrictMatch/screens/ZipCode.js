import React, { Fragment } from 'react';
import { View, Text, TextInput, Image } from 'react-native';
import ButtonSubmit from '../../components/ButtonSubmit';
import PropTypes from 'prop-types';

const zipCodeImage = require('../../../assets/images/district-match-zip.png');

const ZipCode = ({styles, zipcode, error, handleChange, submit}) => {
  return(
    <Fragment>
      <Text style={styles.header}>Just one more thing</Text>
      <Image style={styles.image} source={zipCodeImage}/>
      <Text style={styles.bodyText}>We need your zip code to find your district.</Text>
      <View style={styles.inputContainer}>
        <TextInput
          autoCapitalize="none"
          autoCorrect={false}
          autoFocus={false}
          clearButtonMode="while-editing"
          keyboardType="numeric"
          style={styles.textInput}
          placeholder="ZIP code"
          onChangeText={(value)=>handleChange({zipcode:value})}
          value={zipcode}
        />
        {
          <View>
            <Text style={styles.errorText}>{error && error}</Text>
          </View>
        }
        <ButtonSubmit onPress={submit}/>
      </View>
    </Fragment>
  )
}

ZipCode.propTypes = {
  styles: PropTypes.object,
  zipcode: PropTypes.string,
  error: PropTypes.string,
  handleChange: PropTypes.func,
  submit: PropTypes.func
}

export default ZipCode;
