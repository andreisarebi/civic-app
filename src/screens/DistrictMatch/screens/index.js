export { default as Address } from './Address';
export { default as ZipCode } from './ZipCode';
export { default as Success } from './Success';
export { default as Failure } from './Failure';
