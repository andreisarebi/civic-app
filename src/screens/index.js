import { Navigation } from 'react-native-navigation';
import ActivismScreen from './ActivismScreen';
import ElectionsScreen from './ElectionsScreen';
import FavoritesScreen from './FavoritesScreen';
import HomeScreen from './HomeScreen';
import LoginScreen from './LoginScreen';
import NavigationDrawer from './NavigationDrawer';

export const registerScreens = (store, Provider) => {
  Navigation.registerComponent(
    'civic.ActivismScreen',
    () => ActivismScreen,
    store,
    Provider,
  );
  Navigation.registerComponent(
    'civic.ElectionsScreen',
    () => ElectionsScreen,
    store,
    Provider,
  );
  Navigation.registerComponent(
    'civic.FavoritesScreen',
    () => FavoritesScreen,
    store,
    Provider,
  );
  Navigation.registerComponent(
    'civic.HomeScreen',
    () => HomeScreen,
    store,
    Provider,
  );
  Navigation.registerComponent(
    'civic.LoginScreen',
    () => LoginScreen,
    store,
    Provider,
  );
  Navigation.registerComponent(
    'civic.NavigationDrawer',
    () => NavigationDrawer,
    store,
    Provider,
  );
};
