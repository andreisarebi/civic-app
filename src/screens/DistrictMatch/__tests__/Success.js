import { createSnapshot } from '../../../util/tests';
import { Success } from '../screens';

describe('District Match - Success Screen', () => {
  it('renders correctly', () => {
    const snapshot = createSnapshot(Success, {district:'WA-03'});
    expect(snapshot).toMatchSnapshot();
  });
});
