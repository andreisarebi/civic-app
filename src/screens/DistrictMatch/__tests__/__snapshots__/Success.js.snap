// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`District Match - Success Screen renders correctly 1`] = `
<UNDEFINED>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Your District Is: 
    WA-03
  </Text>
  <Button
    onPress={[Function]}
    title="Go to Elections"
  />
</UNDEFINED>
`;
